src/typechecker.erl

```diff
@@ -747,7 +747,8 @@ normalize({op, _, _, _Arg1, _Arg2} = Op, _TEnv) ->
 normalize({type, Ann, range, [T1, T2]}, TEnv) ->
     {type, Ann, range, [normalize(T1, TEnv), normalize(T2, TEnv)]};
 normalize({type, Ann, map, Assocs}, TEnv) when is_list(Assocs) ->
-    {type, Ann, map, [normalize(As, TEnv) || As <- Assocs]};
+    MTy = {type, Ann, map, [normalize(As, TEnv) || As <- Assocs]},
+    typelib:remove_pos(MTy);
```

Comment:
Hmmm, I'll check this one again.

Reply:
I checked this again and if I remove the commit introducing this change error from the commit message comes back. Maybe this position info removal has to be done somewhere else, but it's definitely needed.

Reply:
Ok, I think https://github.com/josefs/Gradualizer/pull/346/commits/ca9da3c0505bce94e0b511452b3d0e952e494b96 removes the position in a more suitable place. The error seems to be gone, too, so it looks good.

Reply:
In refinable, well, there we also don't do this for any other types. We should be able to find the source of it, i.e. how it manages to avoid remove_pos where the other typea are remove_pos'ed.

If we keep it here, at least add a TODO comment.

Reply:
Hmmm.... I've looked through all the places where `remove_pos` is called and I don't see why maps wouldn't match. AFAICT, remove_pos cascades to all other types from being first called on some function type, since that's what we start type checking from (the top-level functions of a module to be checked) - and in most cases for each `typechecker` function it's only the clause handling function types that calls `remove_pos`.

Moreover, it's not that maps always have the position info when reaching `refinable` - they usually don't:

```
 $ ./bin/gradualizer -pa test_data/ -- test/should_fail/exhaustive_refinable_map_variants.erl
{trace,<0.9.0>,call,
    {typechecker,refinable,
        [{type,0,union,
             [{type,0,map,
                  [{type,0,map_field_exact,
                       [{atom,0,field_one},{type,0,integer,[]}]}]},
              {type,0,map,
                  [{type,0,map_field_exact,
                       [{atom,0,field_two},
                        {type,0,list,[{type,0,char,[]}]}]}]}]},
                        ...
```

For now, I'll just leave a TODO 🤷

Reply:
Nice TODO. Have you managed to trigger this in a test case? If not, it will be hard to know when the TODO is solved.

Reply:
Not in a test case - that's one of the problems - but the original commit 16319c6037704cffcc73e8f5641f79d1d899fc2d has an error report.

Reply:
At least it doesn't do any harm. Let's merge it!

Reply:
Awesome, let's do it 🎉